### ISSUE TITLE
nanoclone is not a function

### ISSUE THREAD

**Issue by Grawl** [nikolay-govorov/nanomerge#1]
I am using `nanomerge` with [Symfony Encore](https://www.npmjs.com/package/@symfony/webpack-encore) like this:

```js
import 'magnific-popup'
import nanomerge from 'nanomerge'
import {
	localization as magnificPopupLocalization,
} from './scripts/vendor-magnific-popup-ru'
const merge = nanomerge
$(document).ready(() => {
	initMagnificPopup()
})
function initMagnificPopup() {
	$.magnificPopup.defaults = merge(
		$.magnificPopup.defaults,
		magnificPopupLocalization, {
			fixedContentPos: true,
			removalDelay: 500,
			mainClass: 'mfp-fade',
		}
	)
}
```

And got an error:

```
index.js:115 Uncaught TypeError: nanoclone is not a function
    at merger (index.js:115)
    at index.js:51
    at Array.forEach (<anonymous>)
    at Object.deep (index.js:50)
    at merger (index.js:126)
    at index.js:133
    at Array.reduceRight (<anonymous>)
    at index.js:132
    at wrapper (index.js:146)
    at initMagnificPopup (script.js:28)
```

Tried to install `nanoclone` into my project but it does not help me.

If I replace `merge` with `lodash.merge`, there is no errors and my code is working like expected.

```js
import lodashMerge from 'lodash.merge'
const merge = lodashMerge
```

**Comment by Grawl** [comment]
previously, this code is working when I use Gulp to build JS.

**Comment by nikolay-govorov** [comment]
Hey @Grawl. Sorry I didn't answer the issue. Is the error still there?

**Comment by Grawl** [comment]
No. Now it's 

```TypeError: nanoclone is not a function. (In 'nanoclone(a)', 'nanoclone' is an instance of Object)```

with the same code

**Comment by mesqueeb** [comment]
@nikolay-govorov I digged deeper into the depths, and it seems if you just `require('nanoclone')` you get the es module with a property called 'default'. So when you do `require('nanoclone')`.default it works without fail!

<img width="261" alt="image" src="https://user-images.githubusercontent.com/3253920/42121433-fe747100-7c69-11e8-8742-4e1ebd375dab.png">

**Comment by nikolay-govorov** [comment]
@mesqueeb, @Grawl, the problem is in the nanoclone package. This occurs when you use the package along with the ES6 syntax. We agreed with the author and within 2-3 days he will correct.

**Comment by nikolay-govorov** [comment]
Hey! I'm released new version and fixed working from webpack. I think it's time to update)

**Comment by mesqueeb** [comment]
Are you sure? There's no updates on the [Nanoclone package](https://github.com/Kelin2025/nanoclone)...

**Comment by nikolay-govorov** [comment]
No, but the problem is that the nanoclone package is built by webpack for node and comes in its original form for connection to collectors. The problem is that it doesn't work isomorphic, and for nanoclone I'm clearly saying that I want to work with a commonJS file

**Comment by mesqueeb** [comment]
Why don't you ask your friend to prepare a Rollup CJS build?

**Comment by mesqueeb** [comment]
I would love nanomerge not to use nanoclone as a dependancy. It might cause unsuspected [bugs](https://github.com/Kelin2025/nanoclone/issues/8)...

**Comment by nikolay-govorov** [comment]
The fact that I have not tested the package with webpack is **my fault**. We found a bug and made a _quick but reliable fix_ so that you can use the package again as soon as possible.

Now I am studying in detail the differences and compatibility of CommonJS and ES6 modules. A little later I will make a patch with a more beautiful and reliable solution.

If there are other problems with nano clone we will solve them and maybe give up this package, but I think that now throw out a working tool for fear – a bad idea. I wouldn't want to solve a problem that doesn't exist yet.

Perhaps there are other pitfalls that I do not see? Also, if you are willing to offer a better solution, I will be happy to accept it.

**Comment by nikolay-govorov** [comment]
Since we fixed the bug, I will close this issue, and I would like to continue the discussion of potential bugs with nano clone in a separate [branch](https://github.com/nikolay-govorov/nanomerge/issues/3).

**Comment by mesqueeb** [comment]
@nikolay-govorov 
I just updated to the last version. It seems npm still has the old build?
https://npm.runkit.com/nanomerge/index.js?t=1530615931801

I get the same error as before.

**Comment by nikolay-govorov** [comment]
No, that's right. I rolled out the re minor version (0.2.0-0), but it NPM ignores it when updating, I consider the version unstable. I ran additional tests and published out the 0.2.0 version so as not to force you to use unstable versions of packages

**Comment by mesqueeb** [comment]
@nikolay-govorov Cool it's on 0.2.0 now! I'll update it in my projects!
